src/calls.rs

```diff
@@ -369,5 +371,45 @@ impl Context {
     }
 }
 
+/// ICE server for JSON serialization.
+#[derive(Serialize, Debug, Clone, PartialEq)]
+struct IceServer {
+    /// STUN or TURN URLs.
+    pub urls: Vec<String>,
+
+    /// Username for TURN server authentication.
+    pub username: Option<String>,
+
+    /// Password for logging into the server.
+    pub credential: Option<String>,
+}
+
+/// Returns JSON with ICE servers.
+///
+/// <https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection#iceservers>
+pub async fn ice_servers(context: &Context) -> Result<String> {
+    let hostname = "ci-chatmail.testrun.org";
+    let port = 3478;
+    let username = "ohV8aec1".to_string();
+    let password = "zo3theiY".to_string();
+
+    // Do not use cache because there is no TLS.
+    let load_cache = false;
+    let urls: Vec<String> = lookup_host_with_cache(context, hostname, port, "", load_cache)
```

Comment:
```suggestion
    let load_cache = false;
    // The primary point of DNS lookup is that Delta Chat desktop
    // relies on the servers being specified by IP,
    // because it itself cannot utilize DNS. See
    // https://github.com/deltachat/deltachat-desktop/issues/5447.
    // This also adds resilience against DNS outages.
    let urls: Vec<String> = lookup_host_with_cache(context, hostname, port, "", load_cache)
```